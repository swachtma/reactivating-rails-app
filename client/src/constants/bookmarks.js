export const SET_BOOKMARK_LOCATIONS = "SET_BOOKMARK_LOCATIONS";
export const SET_BOOKMARK_OFFERED = "SET_BOOKMARK_OFFERED";