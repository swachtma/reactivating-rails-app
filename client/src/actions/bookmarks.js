import * as constants from "../constants/bookmarks";

export const setBookmarkLocations = (chapter_id) => ({
  type: constants.SET_BOOKMARK_LOCATIONS, payload: {chapter: chapter_id}
});

export const setBookmarkOffered = (bool) => ({
  type: constants.SET_BOOKMARK_OFFERED, payload: {prompt_shown: bool}
});