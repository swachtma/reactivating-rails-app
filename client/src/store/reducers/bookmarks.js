import * as constants from '../../constants/bookmarks'

const default_state = {
  prompt_shown: true,
  last_read: { chapter: 1 },
  furthest_read: { chapter: 1 }
}

export const buildLocations = (state, new_loc) => {
  let position = state.furthest_read.chapter < new_loc.chapter ? new_loc : state.furthest_read
  return Object.assign({}, state, { last_read: new_loc, furthest_read: position })
}

export default function bookmarksReducer (state = default_state, action) {
  switch (action.type) {
    case (constants.SET_BOOKMARK_LOCATIONS):
      return buildLocations(state, action.payload)
    case (constants.SET_BOOKMARK_OFFERED):
      return Object.assign({}, state, { prompt_shown: action.payload.prompt_shown })
    default:
      return state
  }
}
