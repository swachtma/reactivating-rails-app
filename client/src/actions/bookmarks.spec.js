/* global expect */
import * as actions from './bookmarks';

describe("Bookmarks actionCreators", ()=>{
  describe("setBookmarkLocations(chapter_id)", ()=>{
    test("matches snapshot", () =>{
      expect(actions.setBookmarkLocations(1)).toMatchSnapshot();
    });
  });
  
  describe("setBookmarkOffered()",()=>{
    test("matches snapshot", () =>{
      expect(actions.setBookmarkOffered()).toMatchSnapshot();
    });
  });
});