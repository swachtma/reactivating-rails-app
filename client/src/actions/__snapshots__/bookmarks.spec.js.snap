// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bookmarks actionCreators setBookmarkLocations(chapter_id) matches snapshot 1`] = `
Object {
  "payload": Object {
    "chapter": 1,
  },
  "type": "SET_BOOKMARK_LOCATIONS",
}
`;

exports[`Bookmarks actionCreators setBookmarkOffered(true) matches snapshot 1`] = `
Object {
  "payload": Object {
    "prompt_shown": true,
  },
  "type": "SET_BOOKMARK_OFFERED",
}
`;
